src/extension.ts

```diff
@@ -304,21 +306,27 @@ export function activate(context: vscode.ExtensionContext) {
     return openUrl(urls[service]);
   };
 
-  const reset = async (newProvider?: string) => {
-    manager.clearAll();
+  const reset = async (initialState?: InitializeState) => {
+    const managerState = manager.getInitialState();
+
+    if (managerState.provider === initialState.provider) {
```

Comment:
this seems like a really bad idea -- why should adding a workspace happen inside `reset`?